### ISSUE TITLE
check `/.well-known/matrix/client`

### ISSUE THREAD

**Issue by richvdh** [matrix-org/matrix-federation-tester#105]
I'm not sure if scope-creeping the federation tester into a general "is my server behaving sanely" tester is something we should do, but:

We could usefully check that `https://<server_name>/.well-known/matrix/client` is sensible (for instance: does the `m.homeserver` point to a working client-server API?)

**Comment by babolivier** [comment]
I believe this is scope-creeping, as the federation tester should only test federation imho. Having a "is my hs all right" tool would definitely be helpful, but I'm not sure this is the right place to do it. Maybe we could have a `matrix-client-tester` project (or some other name of the same essence) that would perform some basic diagnostic on the CS API of a HS?

**Comment by richvdh** [comment]
> I believe this is scope-creeping

well yes, hence my original comment "I'm not sure if scope-creeping ... is something we should do" :)

I guess I've got a couple of thoughts on this:

 * given that a "check my HS is ok" tool is useful, why not build it in here, and rename the project to "matrix-server-tester"? What advantage do we get from keeping the tools separate? I'm actually not sure there is a whole lot we *can* check about the C-S API without trying to log in, and at that point it feels like you're better off just pointing riot-web at the instance.
 * in any case, I'd like to plead client well-known as a bit of a special case, since unlike most of the rest of the C-S API, it operates on the `server_name` of the instance.

**Comment by babolivier** [comment]
> well yes, hence my original comment "I'm not sure if scope-creeping ... is something we should do" :)

Yes, sorry I didn't make it obvious enough that I was replying to that :p

>  that a "check my HS is ok" tool is useful, why not build it in here, and rename the project to "matrix-server-tester"?

It's one possibility, we could probably change this project to be a broader HS tester, with tests on well-known client resolution and also some info-level stuff about e.g. available UIA flows or unstable features.

**Comment by danger89** [comment]
When checking the matrix/client (containing the `base_url`) we can also validate the `public_baseurl`? I think we are all on the same line.

**Comment by AndrewKvalheim** [comment]
Validating [`org.matrix.msc3575.proxy`](https://github.com/matrix-org/matrix-spec-proposals/blob/kegan/sync-v3/proposals/3575-sync.md#unstable-prefix) would also be handy.
